File: cut_tool.py
from pathlib import Path
from check_file_permission import check_file_permission

def cut_file(file_path, delimiter, fields, config):
    if not check_file_permission(file_path, "read", config):
        raise PermissionError(f"Read access denied for {file_path}")
    file_path = Path(file_path).resolve()
    if not file_path.is_file():
        raise ValueError(f"File not found: {file_path}")
    result = []
    with open(file_path, "r", encoding="utf-8") as f:
        for line in f:
            parts = line.strip().split(delimiter)
            selected = [parts[i-1] for i in fields if i <= len(parts)]
            result.append(delimiter.join(selected))
    return "\n".join(result)

File: ts-tools/cut_tool.ts
/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function cutFile(
  filePath: string,
  delimiter: string,
  fields: number[],
  config: any,
): Promise<string> {
  return new Promise((resolve, reject) => {
    checkFilePermission(filePath, 'read', config)
      .then(() => {
        const resolvedPath = path.resolve(filePath);
        fs.readFile(resolvedPath, 'utf-8', (err, data) => {
          if (err) {
            reject(err);
          } else {
            const lines = data.split('\n');
            const result: string[] = [];
            for (const line of lines) {
              const parts = line.split(delimiter);
              const selected = fields.map((i) => parts[i - 1]).filter(Boolean);
              result.push(selected.join(delimiter));
            }
            resolve(result.join('\n'));
          }
        });
      })
      .catch(reject);
  });
}
